test_integration/02_stake_and_mint/utils/progress_mint_assertion.js

```diff
@@ -218,14 +221,22 @@ class ProgressMintAssertion {
     );
     assert.strictEqual(
       eventData._proofProgress,
-      false,
+      proofProgress,
       'Proof progress flag should be false.',
     );
-    assert.strictEqual(
-      eventData._unlockSecret,
-      stakeRequest.unlockSecret,
-      'Unlock secret must match.',
-    );
+    if (proofProgress === true) {
+      assert.strictEqual(
+        eventData._unlockSecret,
+        Utils.ZERO_BYTES32,
```

Comment:
How do you know here that the secret was zero bytes? Doesn't that depend on where this method was called from? Should probably be a function argument here. What do you think?

Or rather: why is the unlock secret in the event and zero when proof is progressed?

Reply:
> How do you know here that the secret was zero bytes? Doesn't that depend on where this > > method was called from? Should probably be a function argument here. What do you think?

`proofProgress` is already a param in the function argument, its value depends on where this method is called from.  it is used to assert different values. 

> Or rather: why is the unlock secret in the event and zero when proof is progressed?

In this way, the facilitator will subscribe to a single event.
If the `progressXXXwithProof` is called, in that case, the Merkle proof is used instead of `unlockSecret`. The event name and params emitted are the same in both `progressXXX` and `progressXXXwithProof` function, so the unlockSecret value will be bytes32(0) as there is no unlock secret. Facilitators can read `_proofProgress` param value to determine if the progress was with the unlock secret or proof data.